File: Vidly/Models/Movie.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace Vidly.Models
{
    //test
    public class Movie
    {
        public int Id { get; set; }

        [Required]
        [StringLength(255)]
        public string Name { get; set; }

        public Genre Genre { get; set; }


        [Required]
        [Display(Name = "Genre")]
        public byte GenreId { get; set; }

        public DateTime DateAdded { get; set; }

        [Required]
        [Display(Name = "Realease Date")]
        public DateTime ReleaseDate { get; set; }

        [Range(1,20)]
        [Required]
        [Display(Name = "Number in Stock")]
        public byte NumberInStock { get; set; }
        public byte NumberAvailable { get; set; }

    }
}

File: Vidly/MyApp/app/model/Movie.js
Ext.define('MyApp.model.Movie', {
    extend: 'Ext.data.Model',
    fields: [
        { name: 'id', type: 'int' },
        { name: 'name', type: 'string' },
        { name: 'genreId', type: 'int' },
        { name: 'dateAdded', type: 'string' },
        { name: 'releaseDate', type: 'string' },
        { name: 'numberInStock', type: 'int' },
        { name: 'numberAvailable', type: 'int' }
    ],
    hasOne: [{
        name: 'genre',
        model: "Genre",
        associationKey: 'genre'
    }]
});